script/foundry/utils/DeployHelper.sol

```diff
@@ -500,13 +500,13 @@ contract DeployHelper is Script, BroadcastManager, JsonDeploymentHandler, Storag
         _postdeploy("IpRoyaltyVaultBeacon", address(ipRoyaltyVaultBeacon));
 
         _predeploy("CoreMetadataModule");
+        impl = address(new CoreMetadataModule(address(accessController), address(ipAssetRegistry)));
         coreMetadataModule = CoreMetadataModule(
-            create3Deployer.deploy(
+            TestProxyHelper.deployUUPSProxy(
+                create3Deployer,
                 _getSalt(type(CoreMetadataModule).name),
-                abi.encodePacked(
-                    type(CoreMetadataModule).creationCode,
-                    abi.encode(address(accessController), address(ipAssetRegistry))
-                )
+                impl,
+                abi.encodeCall(CoreMetadataModule.initialize, address(protocolAccessManager))
             )
```

Comment:
It would be better add check to make sure the poxy and impl deployed properly:

https://github.com/storyprotocol/protocol-core-v1/blob/e339f0671c9172a6699537285e32aa45d4c1b57b/script/foundry/utils/DeployHelper.sol#L232

```solidity
require(
    _getDeployedAddress(type(IPAssetRegistry).name) == address(ipAssetRegistry),
    "Deploy: IP Asset Registry Address Mismatch"
);
require(_loadProxyImpl(address(ipAssetRegistry)) == impl, "IPAssetRegistry Proxy Implementation Mismatch");
impl = address(0); // Make sure we don't deploy wrong impl
```